lib/Parser/Differentiation.pm

```diff
@@ -39,7 +39,11 @@ sub Parser::D {
 		return (0 * $self)->reduce('0*x' => 1) unless defined $self->{variables}{$x};
 		$f = $f->D($x);
 	}
-	return $self->new($f);
+	my $deriv = $self->new($f);
+	$deriv->{string} = $deriv->{tree}->string;
```

Comment:
This line means that any constants are clipped to 6 significant digits (the default number format), so the result may have different from the original.

You may also want to add a new reduction rule that controls doing this doing this, so it can be turned off, if desired.